tests/serial/time/implicit/serial-coupling/WaveformSubsyclingWithConstantAcceleration.cpp

```diff
@@ -0,0 +1,135 @@
+
+  typedef double (*DataFunction)(double);
+
+  DataFunction dataOneFunction = [](double t) -> double {
+    return (double) (2 * t);
+  };
+  DataFunction dataTwoFunction = [](double t) -> double {
+    return (double) (3 * t);
+  };
+  DataFunction writeFunction;
+  DataFunction readFunction;
+
+  std::string meshName, writeDataName, readDataName;
+  if (context.isNamed("SolverOne")) {
+    meshName      = "MeshOne";
+    writeDataName = "DataOne";
+    writeFunction = dataOneFunction;
+    readDataName  = "DataTwo";
+    readFunction  = dataTwoFunction;
+  } else {
+    BOOST_TEST(context.isNamed("SolverTwo"));
+    meshName      = "MeshTwo";
+    writeDataName = "DataTwo";
+    writeFunction = dataTwoFunction;
+    readDataName  = "DataOne";
+    readFunction  = dataOneFunction;
+  }
+
+  double   writeData = 0;
+  double   readData  = 0;
+  double   v0[]      = {0, 0, 0};
+  VertexID vertexID;
+  vertexID = precice.setMeshVertex(meshName, v0);
+
+  int    nSubsteps          = 7; // perform subcycling on solvers. 4 steps happen in each window.
+  int    nWindows           = 5; // perform 5 windows.
+  int    timestep           = 0;
+  double time               = 0;
+  int    timestepCheckpoint = timestep;
+  double timeCheckpoint     = time;
+  int    iterations         = 0;
+
+  if (precice.requiresInitialData()) {
+    writeData = writeFunction(time);
+    precice.writeData(meshName, writeDataName, {&vertexID, 1}, {&writeData, 1});
+  }
+
+  precice.initialize();
+  double maxDt     = precice.getMaxTimeStepSize();
+  double windowDt  = maxDt;
+  double dt        = windowDt / nSubsteps; // Timestep length desired by solver. E.g. 7 steps  with size 1/7
+  double currentDt = dt;                   // Timestep length used by solver
+
+  while (precice.isCouplingOngoing()) {
+    if (precice.requiresWritingCheckpoint()) {
+      timeCheckpoint     = time;
+      timestepCheckpoint = timestep;
+      iterations         = 0;
+    }
+
+    double factor = 1 - pow((1 - 0.1), iterations); // The total relaxation factor over all iterations
+    precice.readData(meshName, readDataName, {&vertexID, 1}, maxDt, {&readData, 1});
+    if (context.isNamed("SolverOne")) { // in the first iteration of each window, use data from previous window.
+      if (iterations == 0) {
+        BOOST_TEST(readData == readFunction(timeCheckpoint));
+      } else {
+        BOOST_TEST(readData == factor * readFunction(time + maxDt));
+      }
+    } else { // in the following iterations, use data at the end of window.
+      BOOST_TEST(readData == readFunction(time + maxDt));
+    }
+
+    precice.readData(meshName, readDataName, {&vertexID, 1}, currentDt, {&readData, 1});
+    if (context.isNamed("SolverOne")) { // in the first iteration of each window, use data from previous window.
+      if (iterations == 0) {
+        BOOST_TEST(readData == readFunction(timeCheckpoint));
+      } else {
+        BOOST_TEST(readData == factor * readFunction(time + currentDt));
```

Comment:
If I merge this PR into my PR #1671 I actually get a failure here. This is good, because it means that your test is sensitive. Of course, it it also to some degree bad, because it means that there is a problem with the implementation of acceleration schemes and waveforms at the moment.